src/DonationForms/Actions/ConvertDonationFormBlocksToFieldsApi.php

```diff
@@ -78,17 +82,24 @@ static function ($node) {
             $form->append($section);
         }
 
-        return $form;
+        return [$form, $this->blockNodeRelationships];
     }
 
     /**
      * @since 3.0.0
      */
     protected function convertTopLevelBlockToSection(BlockModel $block, int $blockIndex): Section
     {
-        return Section::make($block->getShortName() . '-' . $blockIndex)
+        $node = Section::make($block->getShortName() . '-' . $blockIndex)
             ->label($block->getAttribute('title'))
             ->description($block->getAttribute('description'));
+
+        $this->blockNodeRelationships[$block->clientId] = [
+            'block' => $block,
+            'node' => $node,
+        ];
```

Comment:
@pauloiankoski Let's either escalate this to `__invoke` or create a `mapBlockToNode` private function to preserve consistency.

Reply:
d7f8074c972554cc24c31ff497df3c97a0f6e629

Extracted the duplicated code to a private function. Also, rearranged the order of the args of `mapGenericBlockAttributesToNode` so they reflect better the order (block then node).